src/app/transfer-funds/transfer-funds.component.ts

```diff
@@ -401,26 +402,19 @@ export class TransferFundsComponent implements AfterContentInit, OnInit {
       donation.homeBuildingNumber = undefined;
     }
 
-    if (this.personId) {
+    if (this.donor.id) {
       donation.pspCustomerId = this.identityService.getPspId();
     }
 
-    this.createDonation(donation);
-  }
-
-  /**
-   * Creates a Donation itself. Both success and error callbacks should unconditionally set `creatingDonation` false.
-   */
-  private createDonation(donation: Donation) {
```

Comment:
inlined this function - as a function called just once and containing just a single function call I don't think there was much value in having it separate - and having it separate meant repeating the check for `this.donor` being defined.